Prepare 1.28.0 release

djc opened the pull request:
This is the 1st PR for the `1.28.0` release according to our [release process](https://rust-lang.github.io/rustup/dev-guide/release-process.html#making-a-release).

### Checklist

- [x] Bump version in the manifest.
- [x] Update the changelog.
- [ ] ~~Draft the release blog post~~ (Not required, since we're doing a beta release).

djc committed:
Bump version to 1.28.0

djc committed:
Add changelog for 1.28.0

rami3l commented on CHANGELOG.md:
```diff
@@ -1,5 +1,219 @@
 # Changelog
 
+## [1.28.0] - ?
```
It's also worth mentioning that @ChrisDenton has joined our team in this release cycle :)

djc replied:
Added:

> During this release cycle, [Chris Denton](https://github.com/chrisdenton) joined the team. Chris
has a lot of experience contributing to Windows-related parts of the Rust Project, expertise which
was sorely lacking in existing team members, so we're happy to have him on board to help address
Windows-specific issues.

ChrisDenton commented:
Also https://github.com/rust-lang/rustup/blob/c8eddcb4c1dc9aea8067fa73179ce1791aa24d1d/rustup-init.sh#L36?

rami3l commented on CHANGELOG.md:
```diff
@@ -1,5 +1,219 @@
+
+* Replace remaining `winapi` usage with `windows-sys` by @kennykerr in https://github.com/rust-lang/rustup/pull/3802
+* docs(dev-guide): remove “pushing to master” in the release process by @rami3l in https://github.com/rust-lang/rustup/pull/3789
+* chore(deps): update ubuntu docker tag to v24 by @renovate in https://github.com/rust-lang/rustup/pull/3796
```
Although I agree with you that we'd like to maximally take advantage of GitHub-generated changelog, I still consider all these `renovate` PRs to be a bit too noisy... Of course, few would want to look at the full changelog, so maybe that's not that big of a deal? It's your call.

djc replied:
Fair enough -- I went through and reviewed all PRs that contained `chore` and discarded most of them.

rami3l commented:
> Also
> 
> https://github.com/rust-lang/rustup/blob/c8eddcb4c1dc9aea8067fa73179ce1791aa24d1d/rustup-init.sh#L36
> 
> ?

@ChrisDenton Thanks for pointing it out! That's part of the [plan](https://rust-lang.github.io/rustup/dev-guide/release-process.html), yeah, but it cannot be done in this PR for technical reasons. (It should be done right after this PR though.)

rami3l commented on CHANGELOG.md:
```diff
@@ -1,5 +1,219 @@
+
+- The default download backend has been changed from reqwest with native-tls to reqwest with rustls. [pr#3798]
+  - `RUSTUP_USE_CURL` and `RUSTUP_USE_RUSTLS` can still be used to change the download backend
+    if the new backend causes issues.
```
... and, if anything happens, please report to https://github.com/rust-lang/rustup/issues/3806.

djc replied:
Added:

> If issues do happen, please [let us know](https://github.com/rust-lang/rustup/issues/3806).

rami3l reviewed (approved):
There are a few points I'd like to add, but this is looking wonderful. Many thanks!

djc commented:
> There are a few points I'd like to add, but this is looking wonderful. Many thanks!

Do you mean the addition of Chris to the team? If there are other things you'd like to add, now would be a good time to mention that, I think?

djc commented:
> > There are a few points I'd like to add, but this is looking wonderful. Many thanks!
> 
> Do you mean the addition of Chris to the team? If there are other things you'd like to add, now would be a good time to mention that, I think?

Addressed the issues you've mentioned so far!

0xPoe commented on CHANGELOG.md:
```diff
@@ -1,5 +1,192 @@
+
+- The default download backend has been changed from reqwest with native-tls to reqwest with rustls. [pr#3798]
+  - `RUSTUP_USE_CURL` and `RUSTUP_USE_RUSTLS` can still be used to change the download backend
+    if the new backend causes issues. If issues do happen, please [let us know](https://github.com/rust-lang/rustup/issues/3806).
```
This might not be a significant point. However, we(at least recent three releases) usually prefer placing all links at the end of the document. For example:
[let us know]: https://github.com/rust-lang/rustup/issues/3806

I simply followed the format used in previous changelog sections. That said, I think it's acceptable to have two different styles coexisting.

rami3l replied:
> placing all links at the end of the document

I second this, despite the fact that this PR tries to maximally utilize the GitHub-generated detailed changelog. For the "manual" part, at least, it'd still be better to simply follow the established convention.

0xPoe reviewed (commented):
Thanks!

rami3l commented:
> > > There are a few points I'd like to add, but this is looking wonderful. Many thanks!
> > 
> > 
> > Do you mean the addition of Chris to the team? If there are other things you'd like to add, now would be a good time to mention that, I think?
> 
> Addressed the issues you've mentioned so far!

@djc That has been all. Thanks a lot! If anything happens, we can still modify the changelog during the beta testing period :)

djc commented:
@ChrisDenton did you have any other feedback here? (Since you haven't explicitly approved.)

ChrisDenton reviewed (approved):
👍 from me.

Given the long list of changes, I did want to make sure there wasn't anything else worth calling out. But nothing jumps out at me.

rami3l commented:
> @ChrisDenton did you have any other feedback here? (Since you haven't explicitly approved.)

@djc Just don't forget to update the changelog, since we've made some other changes as well since the last commit on this branch. Of course, it can be made into the actual release PR as well :)

djc commented:
Next step in https://github.com/rust-lang/rustup/pull/4098.
